ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -861,6 +863,27 @@ mkByronTx tx = ByronTx
       -- cardano-ledger#581
     }
 
+{-------------------------------------------------------------------------------
+  Block Fetch integration
+-------------------------------------------------------------------------------}
+
+byronBlockOrEBBMatchesHeader :: Header (ByronBlockOrEBB cfg)
+                        -> ByronBlockOrEBB cfg
+                        -> Bool
+byronBlockOrEBBMatchesHeader blkOrEbbHdr blkOrEbb =
+    case unByronHeaderOrEBB blkOrEbbHdr of
+        Left _ebbHdr -> isEbb (unByronBlockOrEBB blkOrEbb)
+        Right hdr    -> isBlockAndHeaderMatchesBody (unByronBlockOrEBB blkOrEbb) hdr
+  where
+    isEbb (CC.Block.ABOBBoundary _) = True
+    isEbb (CC.Block.ABOBBlock _)    = False
+    --
+    isBlockAndHeaderMatchesBody (CC.Block.ABOBBoundary _) _  = False
+    isBlockAndHeaderMatchesBody (CC.Block.ABOBBlock blk) hdr =
+        case CC.Block.validateHeaderMatchesBody hdr (CC.Block.blockBody blk) of
```

Comment:
Alternatively, you could use `isRight`, up to you.

Reply:
Ha, just noticed that and force pushed :stuck_out_tongue: